src/common/utils/string.utils.ts

```diff
@@ -0,0 +1,3 @@
+export const matchesQuery = (field: string, query: string): boolean => {
+    return query ? field.toLowerCase().includes(query.toLowerCase()) : true;
```

Comment:
maybe leverage type inference and default values to make this less verbose? :)

```ts
export const matchesQuery = (field: string, query = '') => field.toLowerCase().includes(query.toLowerCase())
```

Reply:
Two reasons why I'm doing it like this:
- `query` should always be passed, otherwise I don't see the reason to use this function
- When query is passed, we should only be checking when query is truthy - `.includes('')` will always be true and for other null values we can just act dumb and return `true` anyway - only actual strings should be checked

Reply:
We could do this differently though; catch all non-happy paths and just have the happy path at the bottom

Reply:
The only benefit with not having the default value is the performance gain from not calling `.toLowerCase()` on non-truthy `query` values.

Providing a default value does not make the parameter optional quite like `query?: string` would do. I would argue in this case it makes sense to set the default value to an empty string, that way we don't have to do a sanity check in the function body. But it is fine as it is :)

Reply:
> We could do this differently though; catch all non-happy paths and just have the happy path at the bottom

This sounds better imo